x-pack/plugins/apm/server/routes/services/get_services/get_service_stats.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import {
+  AGENT_NAME,
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+  SERVICE_OVERFLOW_COUNT,
+  TRANSACTION_TYPE,
+} from '../../../../common/es_fields/apm';
+import {
+  TRANSACTION_PAGE_LOAD,
+  TRANSACTION_REQUEST,
+} from '../../../../common/transaction_types';
+import { environmentQuery } from '../../../../common/utils/environment_query';
+import { AgentName } from '../../../../typings/es_schemas/ui/fields/agent';
+import {
+  getDocumentTypeFilterForTransactions,
+  getDurationFieldForTransactions,
+  getProcessorEventForTransactions,
+} from '../../../lib/helpers/transactions';
+import { calculateThroughputWithRange } from '../../../lib/helpers/calculate_throughput';
+import {
+  calculateFailedTransactionRate,
+  getOutcomeAggregation,
+} from '../../../lib/helpers/transaction_error_rate';
+import { serviceGroupQuery } from '../../../lib/service_group_query';
+import { ServiceGroup } from '../../../../common/service_groups';
+import { RandomSampler } from '../../../lib/helpers/get_random_sampler';
+import { APMEventClient } from '../../../lib/helpers/create_es_client/create_apm_event_client';
+
+interface AggregationParams {
+  environment: string;
+  kuery: string;
+  apmEventClient: APMEventClient;
+  searchAggregatedTransactions: boolean;
+  maxNumServices: number;
+  start: number;
+  end: number;
+  serviceGroup: ServiceGroup | null;
+  randomSampler: RandomSampler;
+}
+
+export async function getServiceStats({
+  environment,
+  kuery,
+  apmEventClient,
+  searchAggregatedTransactions,
+  maxNumServices,
+  start,
+  end,
+  serviceGroup,
+  randomSampler,
+}: AggregationParams) {
+  const outcomes = getOutcomeAggregation();
+
+  const metrics = {
+    avg_duration: {
+      avg: {
+        field: getDurationFieldForTransactions(searchAggregatedTransactions),
+      },
+    },
+    outcomes,
+  };
+
+  const response = await apmEventClient.search('get_service_stats', {
+    apm: {
+      events: [getProcessorEventForTransactions(searchAggregatedTransactions)],
+    },
+    body: {
+      track_total_hits: false,
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            ...getDocumentTypeFilterForTransactions(
+              searchAggregatedTransactions
+            ),
+            ...rangeQuery(start, end),
+            ...environmentQuery(environment),
+            ...kqlQuery(kuery),
+            ...serviceGroupQuery(serviceGroup),
+          ],
+        },
+      },
+      aggs: {
+        sample: {
+          random_sampler: randomSampler,
+          aggs: {
+            overflowCount: {
```

Comment:
Let's call this `serviceOverflowCount` everywhere.

```suggestion
            serviceOverflowCount: {
```

Reply:
Actually, should be in camel case:
```suggestion
            service_overflow_count: {
```